nsxt/resource_nsxt_edge_transport_node.go

```diff
@@ -1519,7 +1527,7 @@ func setIPAssignmentInSchema(spec *data.StructValue) (interface{}, error) {
 	return []interface{}{elem}, nil
 }
 
-func setHostSwitchProfileIDsInSchema(hspIDs []model.HostSwitchProfileTypeIdEntry) interface{} {
+func setHostSwitchProfileIDsInSchema(hspIDs []model.HostSwitchProfileTypeIdEntry) []interface{} {
```

Comment:
To double check: this is fixing a different issue than the PR description, right? 
(I'm kind of surprised that it could work before)

Reply:
Why wouldn't it work? `interface{}` can be anything, including `[]interface{}`. And before it wasn't iterated anywhere, just assigned into the schema dicts.
As this PR iterates for validation, it requires this change.

Reply:
I think `interace{}` is so generic that even a slice will satisfy it.
`[]interface{}` is a bit more strongly typed as only a slice will satisfy it.

So the change is probably unrelated, but I think a good thing to add.